rosdistro_reviewer/yaml_changes.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
+# Licensed under the Apache License, Version 2.0
+
+from typing import Any
+from typing import Mapping
+from typing import Optional
+
+from git import Repo
+from rosdistro_reviewer.git_lines import get_added_lines
+from rosdistro_reviewer.yaml_lines import AnnotatedSafeLoader
+import yaml
+
+
+def _contains(needle, haystack) -> bool:
+    if needle is not None:
+        for straw in haystack:
+            if needle.start < straw.stop and needle.stop > straw.start:
```

Comment:
Because of Python's support for comparable types I was actually pretty confused about the semantics of this. Function documentation or type annotations might help describe what this does.

Reply:
Doc block and type annotations added in 6af1152.